central/networkgraph/service/service_impl.go

```diff
@@ -101,24 +100,60 @@ func (s *serviceImpl) GetExternalNetworkEntities(ctx context.Context, request *v
 	}
 
 	query, _ = search.FilterQueryWithMap(query, schema.NetworkEntitiesSchema.OptionsMap)
-	pred, err := netEntityPredFactory.GeneratePredicate(query)
+
+	ets, err := s.entities.GetEntityByQuery(ctx, query)
 	if err != nil {
-		return nil, errors.Wrapf(errox.InvalidArgs, "failed to parse query %q: %v", query.String(), err.Error())
+		return nil, err
 	}
 
-	ret, err := s.entities.GetAllMatchingEntities(ctx, func(entity *storage.NetworkEntity) bool {
-		// Do not respect the graph configuration.
-		if entity.GetScope().GetClusterId() == "" || entity.GetScope().GetClusterId() == request.GetClusterId() {
-			return pred.Matches(entity)
-		}
-		return false
-	})
+	return &v1.GetExternalNetworkEntitiesResponse{
+		Entities: ets,
+	}, nil
+}
+
+func (s *serviceImpl) GetExternalNetworkFlows(ctx context.Context, request *v1.GetExternalNetworkFlowsRequest) (*v1.GetExternalNetworkFlowsResponse, error) {
+	// Temporarily elevate permissions to obtain all network flows in cluster.
+	networkGraphGenElevatedCtx := sac.WithGlobalAccessScopeChecker(context.Background(),
```

Comment:
Don't have enough knowledge about SAC, but just checking, could this be used to get the external entities of a deployment that I don't have access to directly? Or is that checked before we get here?

Reply:
I had put this in primarily to follow similar behaviour to the network graph construction, but I think you're right, it would provide access to everything in the cluster. I'll use `ctx` instead, which, if my understanding is correct, will have the access scope of the request (some local testing of this is working as expected)

Reply:
Don't know about you, but I would greatly appreciate some input here from one of the SAC experts, I think that would be Merlin team? Should we ask them?

Reply:
This has been reviewed by @rhybrillou since then, and it seems that the current implementation of SAC is acceptable.